core/src/lib/region_locking.cc

```diff
@@ -0,0 +1,40 @@
+#include <fcntl.h>
```

Comment:
missing license header, may require mentioning of the original author if the code was copied/imported